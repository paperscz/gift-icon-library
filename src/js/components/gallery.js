const styles = require('@scss/main.scss');

import React, { Component } from 'react';
import Proptypes from 'prop-types';
import { connect } from 'react-redux';
import Loading from './loading';
import { loadIcons, openModal } from '../actions';

class Gallery extends Component {
  static propTypes = {
    loadIcons: Proptypes.func,
    openModal: Proptypes.func,
    gallery: Proptypes.array,
  }

  componentDidMount() {
    this.props.loadIcons();
  }

  loadGallery = (gallery) => {
    return gallery.map((content, i) =>
      <li key={`${i}-${content.name}`}
        onClick={() => this.props.openModal(content)}
      >
        <img
          src={content.svgURL}
          alt={content.name}
          width="45px"
          height="45px"
        />
      </li>
    );
  }

  renderGallery = (gallery) => {
    return (
      <div className={`container ${styles['gallery']}`}>
        <h3>Gallery Icons</h3>
        <ul>
          {this.loadGallery(gallery)}
        </ul>
      </div>
    );
  }

  render() {
    const {
      gallery,
    } = this.props;

    return gallery ? this.renderGallery(gallery) : <Loading />;
  }
}

function mapStateToProps(state) {
  return {
    gallery: state.gallery,
  };
}

export default connect(
  mapStateToProps, { loadIcons, openModal }
)(Gallery);
