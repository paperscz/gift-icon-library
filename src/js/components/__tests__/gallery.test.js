import React from 'react';
import { mount } from 'enzyme';

import { Gallery } from '../gallery';

describe('The Gallery component', () => {
  let component;

  const gallery = [
    {
      "key": "test",
      "svgURL": "test.svg",
      "name": "test name",
    },
  ];

  it('render the gallery component', () => {
    component = mount(
      <Gallery
        loadIcons={() => { return; }}
        openModal={() => { return; }}
        gallery={gallery}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('render the loading instead of the gallery component', () => {
    component = mount(
      <Gallery
        openModal={() => { return; }}
        loadIcons={() => { return; }}
      />
    );
    expect(component).toMatchSnapshot();
  });

});
