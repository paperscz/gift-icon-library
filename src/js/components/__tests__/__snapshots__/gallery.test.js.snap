// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Gallery component render the gallery component 1`] = `
<Gallery
  gallery={
    Array [
      Object {
        "key": "test",
        "name": "test name",
        "svgURL": "test.svg",
      },
    ]
  }
  loadIcons={[Function]}
  openModal={[Function]}
>
  <div
    className="container gallery"
  >
    <h3>
      Gallery Icons
    </h3>
    <ul>
      <li
        key="0-test name"
        onClick={[Function]}
      >
        <img
          alt="test name"
          height="45px"
          src="test.svg"
          width="45px"
        />
      </li>
    </ul>
  </div>
</Gallery>
`;

exports[`The Gallery component render the loading instead of the gallery component 1`] = `
<Gallery
  loadIcons={[Function]}
  openModal={[Function]}
>
  <Loading>
    <div
      className="loading"
    >
      <h3>
        Application is Loading
      </h3>
    </div>
  </Loading>
</Gallery>
`;
